### PR TITLE
feat(font): stylistic sets & other OpenType features

### LINKED ISSUE #1294: Ligature stylistic sets support
Hi,

[From the Fira Code wiki:](https://github.com/tonsky/FiraCode/wiki/How-to-enable-stylistic-sets)
> Stylistic sets allow you to access additional, optional parts of the font on a per-feature basis. In Fira Code, they allow you to enable/disable alternative character variants.

This makes switching font features quick and easy compared installing additional fonts. Would you consider adding this to Onivim2?

### PR DESCRIPTION
**Pull request by zbaylin**
This fixes our current `editor.fontLigatures` flag to be able to take in a string of features, similar to what VSCode has.

I.e. `"editor.fontLigatures": "'ss02'"` enables the second stylistic set.

This is achieved through a custom parser using Angstrom. 

Fixes #1294

### CODE REVIEW

File: src/Core/ConfigurationParser.re [hunk 1]
```diff
@@ -192,6 +192,36 @@ let parseString = (~default="", json) =>
   | _ => default
   };
 
+let parseFontLigatures = json =>
```
Comment (bryphe) [inline src/Core/ConfigurationParser.re:195]: Nice! Thanks for exploring implementing this, @zbaylin

**Comment by bryphe** [comment]
Would you mind updating the docs, here, to describe the new `editor.fontLigatures` setting? https://github.com/onivim/oni2/blob/master/docs/docs/configuration/settings.md#editor

**Comment by zbaylin** [comment]
@bryphe definitely!

**Comment by bryphe** [comment]
Looks great, thanks for adding the docs @zbaylin 👍

**Pull request merged**